akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -754,9 +781,15 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
       publish(latestGossip)
 
       val selfStatus = latestGossip.member(selfUniqueAddress).status
-      if (selfStatus == Exiting)
-        shutdown()
-      else if (talkback) {
+      if (selfStatus == Exiting && !exitingInProgress) {
+        // ExitingCompleted will be received via CoordinatedShutdown to continue
+        // the leaving process. Meanwhile the gossip state is not marked as seen.
+        exitingInProgress = true
+        logInfo("Exiting, starting coordinated shutdown")
+        CoordinatedShutdown(context.system).run()
```

Comment:
I am uncomfortable running this here. Is this for testing only? I mean, modules should not call these left and right as it becomes a shutdown spaghetti. We also need ways to have coordinated shutdown if cluster is not involved.